### ISSUE TITLE
thread leak when database down

### ISSUE THREAD

**Issue by chaeyk** [brettwooldridge/HikariCP#233]
When database is not alive, HikariDataSource constructor throws RuntimeException
because HikariPool constructor throws RuntimeException.

```java
   public HikariPool(HikariConfig configuration, String username, String password)
   {
      ...

      addConnectionExecutor = createThreadPoolExecutor(configuration.getMaximumPoolSize(), "HikariCP connection filler (pool " + configuration.getPoolName() + ")", configuration.getThreadFactory(), new ThreadPoolExecutor.DiscardPolicy());
      closeConnectionExecutor = createThreadPoolExecutor(4, "HikariCP connection closer (pool " + configuration.getPoolName() + ")", configuration.getThreadFactory(), new ThreadPoolExecutor.CallerRunsPolicy());

      long delayPeriod = Long.getLong("com.zaxxer.hikari.housekeeping.periodMs", TimeUnit.SECONDS.toMillis(30L));
      houseKeepingExecutorService = new ScheduledThreadPoolExecutor(1, configuration.getThreadFactory() != null ? configuration.getThreadFactory() : new DefaultThreadFactory("Hikari Housekeeping Timer (pool " + configuration.getPoolName() + ")", true));
      houseKeepingExecutorService.setRemoveOnCancelPolicy(true);
      houseKeepingExecutorService.scheduleAtFixedRate(new HouseKeeper(), delayPeriod, delayPeriod, TimeUnit.MILLISECONDS);

      fillPool(); // this throws RuntimeException
   }
```

when fillPool() failed, ThreadPoolExecutors continue to run, and it never be cleaned up.
I think code should be fixed like this.

```java
try {
    fillPool();
} catch (RuntimeException e) {
    addConnectionExecutor.shutdownNow();
    closeConnectionExecutor.shutdownNow();
    houseKeepingExecutorService.shutdownNow();
    throw e;
}
```

or ... constructor should not throw exception when database is down.
connection can be filled by housekeepingexecutor when db is online.

I tested with jdk1.8.0_25

**Comment by brettwooldridge** [comment]
Good catch, this will be fixed in the next release.

**Comment by chaeyk** [comment]
There was initializationFailFast property.
When I set false, exception is not thrown.
I solved issue.
thanks for your comment.

**Comment by brettwooldridge** [comment]
Yes, this will allow HikariCP to start without error, but your original observation is correct; HikariCP should shutdown executors properly when an initialization failure occurs.

**Comment by brettwooldridge** [comment]
This issue is fixed by commit 2667687
